### PR TITLE
docs: document nitro hooks

### PR DESCRIPTION
**Pull request by huang-julien**
### 🔗 Linked issue
resolves https://github.com/nuxt/framework/issues/6571

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

📖 Hi this PR document nitro render hooks 📖

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 1]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
 ::alert{icon=👉}
 Learn more about  [available lifecycle hooks](/api/advanced/hooks)
 ::
+
+## Nitro Hooks (Runtime)
+
+These hooks are available for [Nitro Plugins](https://nitro.unjs.io/guide/advanced/plugins) to hook into nitro's runtime behavior.
```
Comment (danielroe) [inline docs/content/2.guide/4.going-further/2.hooks.md:51]:
```suggestion
These hooks are available for [Nitro plugins](https://nitro.unjs.io/guide/advanced/plugins) to hook into Nitro's runtime behavior.
```

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 2]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
...
+
+### Usage with Server Plugins
```
Comment (danielroe) [inline docs/content/2.guide/4.going-further/2.hooks.md:53]:
```suggestion
### Usage within a Nitro Plugin
```

File: docs/content/3.api/4.advanced/1.hooks.md [hunk 3]
```diff
@@ -28,3 +28,10 @@ Hook                   | Arguments           | Environment     | Description
 Check the [schema source code](https://github.com/nuxt/framework/blob/main/packages/schema/src/types/hooks.ts#L55) for all available hooks.
 
 :NeedContribution
+
+# Nitro Hooks (runtime, server-side)
```
Comment (danielroe) [inline docs/content/3.api/4.advanced/1.hooks.md:32]:
```suggestion
# Nitro App Hooks (runtime, server-side)
```

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 4]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
 ::alert{icon=👉}
 Learn more about  [available lifecycle hooks](/api/advanced/hooks)
 ::
+
+## Nitro Hooks (Runtime)
```
Comment (danielroe) [inline docs/content/2.guide/4.going-further/2.hooks.md:49]:
```suggestion
## Nitro App Hooks (Runtime)
```

Comment (danielroe) [reply docs/content/2.guide/4.going-further/2.hooks.md:49]: (there are also Nitro build hooks which we can document later)

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 5]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
...
+  });
+
+  nitroApp.hooks.hook('render:response', (response, { event }) => {
+    console.log('render:response', response);
+  });
+});
+```
+
+::alert{icon=👉}
+Learn more about  [available nitro lifecycle hooks](/api/advanced/hooks#nitro-hooks-runtime-server-side)
```
Comment (danielroe) [inline docs/content/2.guide/4.going-further/2.hooks.md:69]:
```suggestion
Learn more about [available Nitro lifecycle hooks](/api/advanced/hooks#nitro-hooks-runtime-server-side)
```

**Review by danielroe — APPROVED** [review]
Thank you ❤️

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 6]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
...
+export default defineNitroPlugin((nitroApp) => {
+  nitroApp.hooks.hook('render:html', (html, { event }) => {
+    console.log('render:html', html);
+    html.bodyAppend.push('<hr>Appended by custom plugin');
+  });
+
+  nitroApp.hooks.hook('render:response', (response, { event }) => {
+    console.log('render:response', response);
+  });
+});
```
Comment (DamianGlowala) [inline docs/content/2.guide/4.going-further/2.hooks.md:65]:
```suggestion
    console.log('render:html', html)
    html.bodyAppend.push('<hr>Appended by custom plugin</hr>')
  });

  nitroApp.hooks.hook('render:response', (response, { event }) => {
    console.log('render:response', response)
  })
})
```

File: docs/content/2.guide/4.going-further/2.hooks.md [hunk 7]
```diff
@@ -45,3 +45,26 @@ export default defineNuxtPlugin((nuxtApp) => {
...
+```
+
+::alert{icon=👉}
+Learn more about [available Nitro lifecycle hooks](/api/advanced/hooks#nitro-hooks-runtime-server-side)
```
Comment (DamianGlowala) [inline docs/content/2.guide/4.going-further/2.hooks.md:69]:
```suggestion
Learn more about available [Nitro lifecycle hooks](/api/advanced/hooks#nitro-hooks-runtime-server-side).
```

File: docs/content/3.api/4.advanced/1.hooks.md [hunk 8]
```diff
@@ -28,3 +28,10 @@ Hook                   | Arguments           | Environment     | Description
 Check the [schema source code](https://github.com/nuxt/framework/blob/main/packages/schema/src/types/hooks.ts#L55) for all available hooks.
 
 :NeedContribution
+
+# Nitro App Hooks (runtime, server-side)
+
+Hook                   | Arguments             | Description                          | Types
+-----------------------|-----------------------|--------------------------------------|------------------
+`render:response`      | `response, { event }` | Called before sending the response.  | [response](https://github.com/unjs/h3/blob/f6ceb5581043dc4d8b6eab91e9be4531e0c30f8e/src/types.ts#L38), [event](https://github.com/unjs/h3/blob/f6ceb5581043dc4d8b6eab91e9be4531e0c30f8e/src/types.ts#L38)
```
Comment (DamianGlowala) [inline docs/content/3.api/4.advanced/1.hooks.md:36]: (unsure whether updated URL is pointing to the right place)

```suggestion
`render:response`      | `response, { event }` | Called before sending the response.  | [response](https://github.com/nuxt/framework/blob/71ef8bd3ff207fd51c2ca18d5a8c7140476780c7/packages/nuxt/src/core/runtime/nitro/renderer.ts#L24), [event](https://github.com/unjs/h3/blob/f6ceb5581043dc4d8b6eab91e9be4531e0c30f8e/src/types.ts#L38)
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

File: docs/content/2.guide/4.going-further/2.hooks.md (hunk 6 above)
Comment (danielroe) [reply docs/content/2.guide/4.going-further/2.hooks.md:65]: `<hr>` is void and so I think it was right before...

File: docs/content/3.api/4.advanced/1.hooks.md (hunk 8 above)
Comment (danielroe) [reply docs/content/3.api/4.advanced/1.hooks.md]: good spot!

File: docs/content/2.guide/4.going-further/2.hooks.md (hunk 6 above)
Comment (DamianGlowala) [reply docs/content/2.guide/4.going-further/2.hooks.md]: 🤦 ofc it is, my bad (: I mistakenly treated it as a header tag for some reason.

**Review by danielroe — APPROVED** [review]

**Review by pi0 — APPROVED** [review]
Thanks <3

**Pull request merged**